ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/OnChainPrivacyPrecompiledContract.java

```diff
@@ -190,10 +191,13 @@ public Bytes compute(final Bytes input, final MessageFrame messageFrame) {
     }
 
     if (privateTransaction.isGroupRemovalTransaction()) {
+      // get first participant parameter - there will be only one for removal transaction
+      final String removedParticipant = getParticipantsFromParameter(privateTransaction.getPayload()).get(0);
+
       final PrivateTransactionEvent removalEvent =
           new PrivateTransactionEvent(
               privateTransaction.getPrivacyGroupId().get().toBase64String(),
-              privateTransaction.getPrivateFrom().toBase64String());
+              removedParticipant);
       privateTransactionEventObservers.forEach(
           sub -> sub.onPrivateTransactionProcessed(removalEvent));
```

Comment:
We could move this to a private method `SendRemovedParticipantEvent()` to try to keep the `compute()` method a bit cleaner (I know it is already kinda messy, but let's try to not make it worse) :)

Reply:
will clean this up today!

Reply:
refactored